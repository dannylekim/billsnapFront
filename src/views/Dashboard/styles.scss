div.dashboard__flexbox {
  display: flex;
  flex-wrap: nowrap;
  width: 100%;
  height: 100vh;
  margin: 0;
  padding: 0;

  div.bill__display {
    height: inherit;
    width: calc(100% - 302px);
    margin-top: 35px;
  }
}

