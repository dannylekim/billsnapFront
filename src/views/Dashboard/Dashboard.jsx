import React, {Component} from "react";
import {Button, Nav, NavItem, NavLink} from "shards-react";

import navItems from "../../constants/BillDisplayNav.json";

import BillDisplay from "../BillDisplay";
import SearchBar from "../../components/SearchBar";
import Loader from "../../components/Loader";
import BillSummary from "../../components/BillSummary";

import "./styles.scss";
import PendingBillsContainer from "../../components/PendingBillsContainer";
import LargeBillCard from "../../components/BillCard/LargeBillCard";

export default class Dashboard extends Component {
  constructor(props) {
    super(props);
    this.state = {
      currentActiveTab: "allBills",
    };
  }

  displayTab = (navType) => {
    switch (navType) {
      case "allBills":
        return <BillDisplay />;
      case "owedToYou":
        return <PendingBillsContainer />;
      default:
        return <Loader />;
    }
  };

  render() {
    const { currentActiveTab } = this.state;
    const { activeBill } = this.props;

    return (
      <>
        {this.props.hasUser ? (
          <div className="dashboard__flexbox">
            <div className="bill__wrapper">
              <div className="bill__section">
                <SearchBar activeTab={currentActiveTab} />
                <Button id="add__bill__button"> {"+ Add bill"} </Button>
                <Nav tabs>
                  {navItems.map((item, key) => (
                    <NavItem key={key}>
                      <NavLink
                        active={this.state.currentActiveTab === item.name}
                        onClick={() =>
                          this.setState({ currentActiveTab: item.name })
                        }
                      >
                        {item.title}
                      </NavLink>
                    </NavItem>
                  ))}
                </Nav>

                <div className="bill__list__section">
                  {this.displayTab(this.state.currentActiveTab)}
                </div>
              </div>
              <div className="specific__bill__section">
                <BillSummary />
                {activeBill && <LargeBillCard selectedBill={activeBill} />}
              </div>
            </div>
          </div>
        ) : (
          this.props.history.push("/")
        )}
      </>
    );
  }
}
