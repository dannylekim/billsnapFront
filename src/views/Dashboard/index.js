import Dashboard from "./Dashboard";
import {connect} from "react-redux";

const mapStateToProps = (state) => ({
  userInfo: state.users.userInfo,
  hasUser: Object.keys(state.users.userInfo).length > 0,
  isBillLoading: state.application.isBillLoading,
});

export default connect(mapStateToProps)(Dashboard);
