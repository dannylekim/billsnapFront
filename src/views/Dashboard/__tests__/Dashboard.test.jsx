import React from "react";
import Dashboard from "../Dashboard";

jest.mock("../../BillDisplay", () => "BillDisplay");
jest.mock(
  "../../../components/PendingBillsContainer",
  () => "PendingBillsContainer"
);
jest.mock("../../../components/BillSummary", () => "BillSummary");
jest.mock("../../../components/SearchBar", () => "SearchBar");

describe("Dashboard", () => {
  describe("render", () => {
    beforeEach(() => {
      localStorage.setItem("billSnap_token", "token");
    });

    afterEach(() => {
      localStorage.clear();
    });

    describe("snapshots ðŸ“¸", () => {
      it("Dashboard should match snap shot", () => {
        matches(<Dashboard history={{ push: jest.fn() }} />);
      });

      it("Dashboard should match snap shot on notifications tab", () => {
        let wrapper;

        wrapper = shallow(<Dashboard history={{ push: jest.fn() }} />);

        wrapper.setState((prev) => ({
          ...prev,
          currentActiveTab: "owedToYou",
        }));

        matches(wrapper);
      });

      it("Dashboard should match snap shot on nothing", () => {
        let wrapper;

        wrapper = shallow(<Dashboard history={{ push: jest.fn() }} />);

        wrapper.setState((prev) => ({
          ...prev,
          currentActiveTab: "",
        }));

        matches(wrapper);
      });
    });
  });
});
