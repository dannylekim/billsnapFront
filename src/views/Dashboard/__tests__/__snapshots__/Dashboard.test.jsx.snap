// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render snapshots ðŸ“¸ Dashboard should match snap shot 1`] = `
<div>
  <div
    className="bill__wrapper"
  >
    <p>
       No Bills Found
    </p>
  </div>
</div>
`;
